### ISSUE TITLE
TypeError: null is not an object (evaluating 'libraryController.importShareableObjectReference_intoDocument(this._object, this._documentData).localObject')

### ISSUE THREAD

**Issue by ivanortegaalba** [BohemianCoding/SketchAPI#178]
Hi :)

I'm working with the new Library implementation in JS API. Working on Sketch 50 beta and the branch to introduce the compatibility on libraries. I have found this problem.

Running my code:

```
import api from './api';

export default function(context) {
    var document = api.getSelectedDocument();
    var libraries = api.getLibraries();

    libraries.forEach((library) => {
        const symbols = library.getImportableSymbolReferencesForDocument(document);

        symbols.forEach((symbol) => {
            symbol.import()
        });
    });
}
```

When I call `import()` I receive this error:

```
TypeError: null is not an object (evaluating 'libraryController.importShareableObjectReference_intoDocument(this._object, this._documentData).localObject')
line: 6251
```
And researching about the implementation this function `importShareableObjectReference_intoDocument` return null:
```js
import() {
    if (!this._documentData) {
        throw new Error('missing document data');
    }
    const libraryController = AppController.sharedInstance().librariesController();
    console.log(
        'importedSymbol',
        libraryController.importShareableObjectReference_intoDocument(
            this._object,
            this._documentData
        ) // -> null
    );

    const importedSymbol = libraryController
    .importShareableObjectReference_intoDocument(this._object, this._documentData)
    .localObject();

    return wrapNativeObject(importedSymbol);
}
```
So, when we call `.localObject()` the error is thrown.

**Comment by mathieudutour** [comment]
Yes indeed, the API changed in between. Thanks for spotting this. 

Other than that, does the API make sense? Any rough edges?

**Comment by ivanortegaalba** [comment]
For me it makes sense.

I'm trying to generate a document with all symbols from libraries. I'm trying to use this new API, and my approach is this:

````
import api from './api';

export default function(context) {
    var document = api.getSelectedDocument();
    var libraries = api.getLibraries(); // Something to select only the user library?

    libraries.forEach((library) => {
        const symbols = library.getImportableSymbolReferencesForDocument(document);

        symbols.forEach((symbol) => {
            const importedSymbol = symbol.import()
            // What would be your approach to have this symbols locally? 
            // We need to take in consideration that some symbols can reference other symbols as override or nestedSymbols. This is contemplated?
        });
    });
}
````

Thanks so much! :)

**Comment by ivanortegaalba** [comment]
Also @mathieudutour, This is blocking for me, Would I have a workaround? Thanks!!!

**Comment by mathieudutour** [comment]
What is “this”?

**Comment by mathieudutour** [comment]
Also this bug only happens when trying to import a symbol that have already been imported

**Comment by ivanortegaalba** [comment]
But this error is thrown when I call the import the first symbol. 
Does it make sense that you receive a symbol already imported from the function `getImportableSymbolReferencesForDocument `?

**Comment by mathieudutour** [comment]
Well it’s probably already imported by another symbol or something. 

Yes it makes sense that it’s returned even if it’s already imported because it’s still a reference to a Library.

**Comment by ivanortegaalba** [comment]
Oh! Thank you @mathieudutour! Now, I have skipped the already imported symbols, because I have nested symbols inside other symbols.

Now we have the imported symbols in the document, but they still are linked to the library. 
Is there a way through the API to unlink these symbols from the library? We are trying to consolidate multiple libraries into a single library, so we could distribute it easily ^^

<img width="454" alt="captura de pantalla 2018-04-20 a las 15 41 03" src="https://user-images.githubusercontent.com/5699976/39054336-a5008694-44b1-11e8-93d8-c36d9a516dd5.png">
